core/src/bufr/BufrReader/Exports/Variables/SensorScanAngleVariable.cpp

```diff
@@ -39,89 +107,58 @@ namespace bufr {
     {
         checkKeys(map);
 
-        // Get input parameters for sensor scan angle calculation
-        std::string sensor;
-        if (conf_.has(ConfKeys::Sensor) )
-        {
-             sensor = conf_.getString(ConfKeys::Sensor);
-        }
-        else
-        {
-            throw eckit::BadParameter("Missing required parameters: sensor. "
-                                      "Check your configuration.");
-        }
+        // Sensor string is required
+        const std::string sensor = conf_.getString(ConfKeys::Sensor);
 
-        float start;
-        float step;
-        float stepAdj;
-        if (conf_.has(ConfKeys::ScanStart) && conf_.has(ConfKeys::ScanStep))
-        {
-             start = conf_.getFloat(ConfKeys::ScanStart);
-             step = conf_.getFloat(ConfKeys::ScanStep);
+        // Required parameters
+        if (!conf_.has(ConfKeys::ScanStart) || !conf_.has(ConfKeys::ScanStep)) {
+            throw eckit::BadParameter("Missing required parameters: scanStart or scanStep. Check configuration.");
         }
-        else
-        {
-            throw eckit::BadParameter("Missing required parameters: scan starting angle and step. "
-                                      "Check your configuration.");
-        }
-
-        if (conf_.has(ConfKeys::ScanStepAdjust) && sensor == "iasi" )
-        {
-             sensor = conf_.getString(ConfKeys::Sensor);
-             stepAdj = conf_.getFloat(ConfKeys::ScanStepAdjust);
+        const float start = conf_.getFloat(ConfKeys::ScanStart);
+        const float step  = conf_.getFloat(ConfKeys::ScanStep);
+
+        // Optional: only needed for IASI
+        float stepAdj = 0.0f;
+        if (sensor == "iasi") {
+            if (!conf_.has(ConfKeys::ScanStepAdjust)) {
+                throw eckit::BadParameter("Missing required parameter: scanStepAdjust for IASI");
+            }
+            stepAdj = conf_.getFloat(ConfKeys::ScanStepAdjust);
         }
 
-        // Read the variables from the map
-
+        // Get required input field: FOV number
         auto& fovnObj = map.at(getExportKey(ConfKeys::FieldOfViewNumber));
+        const size_t nobs = fovnObj->size();
+        auto fovn = std::dynamic_pointer_cast<DataObject<int>>(fovnObj)->getRawData();
 
         // Declare and initialize scanline array
         // scanline has the same dimension as fovn
-        std::vector<float> scanang(fovnObj->size(), DataObject<float>::missingValue());
-        std::vector<int> scanpos(fovnObj->size(), DataObject<int>::missingValue());
+        std::vector<float> scanang(nobs, DataObject<float>::missingValue());
+        std::vector<int> scanpos(nobs);
+        std::vector<int> forn;
 
-        // Get field-of-view number
-        std::vector<int> fovn(fovnObj->size(), DataObject<int>::missingValue());
-        for (size_t idx = 0; idx < fovnObj->size(); idx++)
+        if (sensor == "cris")
```

Comment:
There are a lot of if statements like this in this function.. Would if make sense to have seperate functions, one for CRIS and one for IASI one for Generic and then call them from this function (get rid of the if statements)?

Reply:
OK.  Good suggestion.  I will try that!

Reply:
@rmclaren I modified the code to use fewer' if statements`. Please let me know if I can improve the code in any way. Thanks.